packages/shared-influx-pieces/src/lib/InfluxListener.ts

```diff
@@ -0,0 +1,51 @@
+import type { Point } from '@influxdata/influxdb-client';
+import { Listener } from '@skyra/http-framework';
+import { getClientId } from './api.js';
+import { Tags } from './enum.js';
+import { isInfluxInitialized } from './functions.js';
+
+export abstract class InfluxListener extends Listener {
+	public tags: [Tags, string][] = [];
+
+	public constructor(context: Listener.LoaderContext, options: InfluxListener.Options) {
+		super(context, { enabled: isInfluxInitialized(), ...options });
+	}
+
+	public override async onLoad() {
+		await this.initTags();
+		return super.onLoad();
+	}
+
+	public writePoint(point: Point) {
+		return this.container.analytics!.writeApi.writePoint(this.injectTags(point));
+	}
+
+	public writePoints(points: readonly Point[]) {
+		points = points.map((point) => this.injectTags(point));
+		return this.container.analytics!.writeApi.writePoints(points);
+	}
+
+	public flush() {
+		return this.container.analytics!.writeApi.flush();
+	}
+
+	protected injectTags(point: Point) {
+		for (const tag of this.tags) {
+			point.tag(tag[0], tag[1]);
+		}
+		return point;
+	}
+
+	protected async initTags() {
+		const clientID = await getClientId();
+		if (clientID) this.tags.push([Tags.Client, clientID]);
```

Comment:
```suggestion
		const clientId = await getClientId();
		if (clientId) this.tags.push([Tags.Client, clientId]);
```

Reply:
Also, we could probably expose `container.applicationCommandRegistry.clientId`? cc: @favna 
https://github.com/skyra-project/archid-components/blob/fa92574ad8b9cfec47c0328c79fe3f8d2ac02d9f/packages/http-framework/src/lib/interactions/shared/ApplicationCommandRegistry.ts#L249-L252

Reply:
> Also, we could probably expose `container.applicationCommandRegistry.clientId`? cc: @favna
> 
> https://github.com/skyra-project/archid-components/blob/fa92574ad8b9cfec47c0328c79fe3f8d2ac02d9f/packages/http-framework/src/lib/interactions/shared/ApplicationCommandRegistry.ts#L249-L252

Why not put it on the client directly?
You can retrieve it from this one for this function, right?

Reply:
https://github.com/skyra-project/archid-components/pull/320 is now merged, please implement this instead.